### PR TITLE
vim: add options

### PR DESCRIPTION
**Pull request by uvNikita**

### CODE REVIEW

File: modules/programs/vim.nix [hunk 1]
```diff
@@ -38,7 +46,7 @@ let
     let
       v =
         if isBool value then (if value then "" else "no") + name
-        else name + "=" + toString value;
+        else name + "=" + (if isList value then (concatStringsSep "," value) else toString value);
```
Comment (rycee) [inline modules/programs/vim.nix]: I was thinking how to shorten the length of this line and came up with
```nix
else
  "${name} = ${
    if isList value
    then concatStringsSep "," value
    else toString value
  }";
```

**Comment by rycee** [comment]
Looks good to me :+1: I just added a minor comment which you may or may not want to consider, feel free to merge whenever.

**Comment by teto** [comment]
apologies if I already wrote anything about it (I really dislike the current nixpkgs vim configure system, yet no time to fix it), but if you are really interested going that deep into defining options, you might be able to automate it from https://github.com/neovim/neovim/blob/master/src/nvim/options.lua .

File: modules/programs/vim.nix (hunk 1 above)
Comment (uvNikita) [reply modules/programs/vim.nix]: Good suggestion, thanks.

**Comment by uvNikita** [comment]
@teto I agree that makes sense to automate this part and reuse work made by neovim, but I don't think I will have the possibility to do this myself anytime soon. But thanks for the suggestion anyway.

**Comment by uvNikita** [comment]
This pull request has been rebased via `git hub pull rebase`. Original pull request HEAD was fd5bb1d974ee395a622decd73d41ccac72ea84d3, new (rebased) HEAD is a2e09b4c9dabdf3a4481c51091e472f433b4a0be

**Pull request closed without merging**